scientific/README.md

```diff
@@ -0,0 +1,135 @@
+
+In addition, some units may be multiplied or divided by other units depending on how they are defined. For instance, you can create a Force value by multiplying a `Weight` and `Acceleration` unit. The unit of the returned ScientificValue will be determined based on the input. Calculating with values in the Imperial system generally returns an imperial unit, while using CGS-units will usually return a CGS unit. Alternatively the unit can be explicitly defined by using the creation method (usually named after the `PhysicalQuantity`).
+
+```kotlin
+val weight = 10(Kilogram)
+val acceleration = 3(Meter per Second per Second)
+val force = weight * acceleration // Returns in Newton
+val dyneForce = weight.convert(Gram) * acceleration.convert(Centimeter per Second per Second) // Returns in Dyne
+val poundForce = PoundForce.force(weight, acceleration) // Returns in PoundForce even though constructing units are in metric
+```
+
+### Scientific Array
+
+The `ScientificArray` allows to mark a list of Numbers as values of a `ScientificUnit`. A `DefaultScientificArray` can be easily created using `listOf(1, 2, 3)(Meter)`. Alternatively, create a custom ScientificArray implementation and pass a construction function in the invoke method. Since `ScientificArray` expects all values to be of the same number type, convenience interfaces are created: `DoubleScientificArray`, `IntScientificArray`, `FloatScientificArray`, `LongScientificArray`, `ShortScientificArray`, and `ByteScientificArray`.
+
+```kotlin
+value class HeartRateArray(override val values: DoubleArray) : DoubleScientificArray<PhysicalQuantity.Frequency, BeatsPerMinute> {
+
+    constructor(values: List<Decimal>, unit: ScientificUnit<PhysicalQuantity.Frequency>) : this(values.map { unit.convert(it, BeatsPerMinute) }.toDoubleArray())
+    
+    override val unit: BeatsPerMinute
+        get() = BeatsPerMinute
+}
+
+val defaultBPMArray = listOf(50, 60, 80)(BeatsPerMinute)
+val heartRateArray = listOf(40, 60, 80)(BeatsPerMinute, ::HeartRateArray)
+```
+
+A list of `ScientificValue` of the same `PhysicalQuantity` can also be merged using the `toScientificArray()` method.
+
+```kotlin
+val defaultBPMArray = listOf(50(BeatsPerMinute), 60(BeatsPerMinute), 80(BeatsPerMinute)).toScientificArray(BeatsPerMinute)
+val heartRateArray = listOf(40(BeatsPerMinute), 60(BeatsPerMinute), 80(BeatsPerMinute)).toScientificArray(BeatsPerMinute, ::HeartRateArray)
+```
+
+#### Map, Combine, Concat, and Split
+
+ScientificArray can be modied using some methods:
+
+- `map` takes all values, converts them to a `ScientificUnit` and transforms these using a transformation method, returning a ScientificArray of all resulting units.
+
+  ```kotlin
+  val interval = listOf(1, 1.5, 0.9)(Second)
+  val defaultBPMArray = interval.map { convert(Minute).frequency() }
```

Comment:
How does it convert from Seconds to frequency?

Reply:
same function, frequency()